File: app/src/main/java/com/guideapp/guideapp/sync/GuideFirebaseJobService.java
package com.guideapp.guideapp.sync;

import android.content.Context;
import android.os.AsyncTask;

import com.firebase.jobdispatcher.JobParameters;
import com.firebase.jobdispatcher.JobService;

public class GuideFirebaseJobService extends JobService {
    private AsyncTask<Void, Void, Void> mFetchGuideTask;

    @Override
    public boolean onStartJob(final JobParameters jobParameters) {

        mFetchGuideTask = new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(Void... voids) {
                Context context = getApplicationContext();
                GuideSyncTask.syncGuide(context);
                jobFinished(jobParameters, false);
                return null;
            }

            @Override
            protected void onPostExecute(Void aVoid) {
                jobFinished(jobParameters, false);
            }
        };

        mFetchGuideTask.execute();
        return true;
    }

    @Override
    public boolean onStopJob(JobParameters jobParameters) {
        if (mFetchGuideTask != null) {
            mFetchGuideTask.cancel(true);
        }
        return true;
    }
}

File: app/src/main/java/com/guideapp/sync/GuideFirebaseJobService.kt
package com.guideapp.sync

import android.annotation.SuppressLint
import android.content.Context
import android.os.AsyncTask

import com.firebase.jobdispatcher.JobParameters
import com.firebase.jobdispatcher.JobService

class GuideFirebaseJobService : JobService() {
    private var mFetchGuideTask: AsyncTask<Void, Void, Void>? = null

    @SuppressLint("StaticFieldLeak")
    override fun onStartJob(jobParameters: JobParameters): Boolean {

        mFetchGuideTask = object : AsyncTask<Void, Void, Void>() {
            override fun doInBackground(vararg voids: Void): Void? {
                val context = applicationContext
                GuideSyncTask.syncGuide(context)
                jobFinished(jobParameters, false)
                return null
            }

            override fun onPostExecute(aVoid: Void) {
                jobFinished(jobParameters, false)
            }
        }

        mFetchGuideTask?.execute()
        return true
    }

    override fun onStopJob(jobParameters: JobParameters): Boolean {
        mFetchGuideTask?.cancel(true)
        return true
    }
}